src/main/java/org/opensearch/security/configuration/MaskedField.java

```diff
@@ -152,7 +152,7 @@ private static byte[] customHash(byte[] in, final String algorithm) {
     private byte[] blake2bHash(byte[] in) {
         // Salt is passed incorrectly but order of parameters is retained at present to ensure full backwards compatibility
         // Tracking with https://github.com/opensearch-project/security/issues/4274
-        final Blake2b hash = new Blake2b(null, 32, null, defaultSalt);
+        final Blake2b hash = new Blake2b(null, 32, defaultSalt, null);
```

Comment:
Hi @prabhask5 I had made the same code change in my original PR but concerns were raised that some customers may be reliant on certain hash values, even though generated by this incorrect call, so I backed the change out and raised the bug to discuss the best approach going forward.

One possibility here would be to add the fix but wrap it in an if statement that uses the newer, correct code from OpenSearch version 3.x onwards. It could then also be documented as a potential breaking change for some customers. What do you think?

Note I also had to change the corresponding tests that were reliant on this hashing in my original PR.